src/runtime/virtcontainers/qemu_arch_base.go

```diff
@@ -142,8 +143,33 @@ type qemuArch interface {
 
 	// append pvpanic device
 	appendPVPanicDevice(devices []govmmQemu.Device) ([]govmmQemu.Device, error)
+
+	// append protection device.
+	// This implementation is architecture specific, some archs may need
+	// a firmware, returns a string containing the path to the firmware that should
+	// be used with the -bios option, ommit -bios option if the path is empty.
+	appendProtectionDevice(devices []govmmQemu.Device, firmware string) ([]govmmQemu.Device, string, error)
```

Comment:
As discussed in today's use case meeting, the SEV device requires a change to the VM configuration that isn't exposed in these current arguments. It was recommended that we change this interface to the following:
 `func (q *qemuAmd64) appendProtectionDevice( govmmQemu.Config ) ( govmmQemu.Config, error ) {...}`

Reply:
this a weird return value for a function that appends a device, other `append*Device` functions return an `[]govmmQemu.Device`, why you need a `govmmQemu.Config` ?

Reply:
I have commented in your govmm PR, please take a look and let me know what do you think, maybe we should append a `s` to the name of this function (`appendProtectionDevices`)

Reply:
Yes, I agree. In this way the `sevProtection` case will add the `govmmQemu.Guest` and `govmmQemu.OVMF` devices, removing the need for an OVMF knob. `appendProtectionDevices` sounds like an appropriate change :)

Reply:
@devimc Thinking it over a bit more... Since the SEV object has a number of arch-specific fields (e.g., `CbitPos`, `ReducePhyBits`..and growing). I think what makes the most sense is to create a `govmmQemu.SevObject` which encompasses both the SEV object and OVMF drive. This also means the interface can stay `appendProtectionDevice`.

WDYT?

Reply:
> @devimc Thinking it over a bit more... Since the SEV object has a number of arch-specific fields (e.g., CbitPos, ReducePhyBits..and growing). I think what makes the most sense is to create a govmmQemu.SevObject which encompasses both the SEV object and OVMF drive. This also means the interface can stay appendProtectionDevice.

@jimcadden sounds good to me :+1: